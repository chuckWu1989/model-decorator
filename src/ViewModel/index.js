import React, { Component } from 'react';

const ViewModel = Model => (
  WrappedComponent => (
    class extends Component {
      componentWillMount() {
        this.setState({ model: new Model() });
      }
      render() {
        const { props, state: { model } } = this;
        const modelProps = model.setValues(props);
        return (
          <WrappedComponent {...props} {...modelProps} />
        );
      }
    }
  )
);

export default ViewModel;
