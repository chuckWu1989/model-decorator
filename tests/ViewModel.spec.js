import React from 'react';
import toJson from 'enzyme-to-json';
import ViewModel from '../src/ViewModel';

const { shallow } = global;

describe('ViewModel', () => {
  it('should render properly', () => {
    const model = function Model() {
      this.setValues = () => {};
    };
    const WrappedComponent = () => <div />;
    const ViewModelComponent = ViewModel(model)(WrappedComponent);
    const tree = toJson(shallow(<ViewModelComponent />));
    expect(tree).toMatchSnapshot();
  });
});
