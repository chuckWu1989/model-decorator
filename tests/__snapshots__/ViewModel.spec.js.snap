// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewModel should render properly 1`] = `
<WrappedComponent
  model={
    Model {
      "setValues": [Function],
    }
  }
/>
`;
